controllers/component_controller_conditions.go

```diff
@@ -28,80 +28,151 @@ import (
 	logutil "github.com/redhat-appstudio/application-service/pkg/log"
 )
 
-func (r *ComponentReconciler) SetCreateConditionAndUpdateCR(ctx context.Context, req ctrl.Request, component *appstudiov1alpha1.Component, createError error) {
+func (r *ComponentReconciler) SetCreateConditionAndUpdateCR(ctx context.Context, req ctrl.Request, component *appstudiov1alpha1.Component, createError error) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	condition := metav1.Condition{}
 	if createError == nil {
-		meta.SetStatusCondition(&component.Status.Conditions, metav1.Condition{
+		condition = metav1.Condition{
 			Type:    "Created",
 			Status:  metav1.ConditionTrue,
 			Reason:  "OK",
 			Message: "Component has been successfully created",
-		})
+		}
 	} else {
-		meta.SetStatusCondition(&component.Status.Conditions, metav1.Condition{
+		condition = metav1.Condition{
 			Type:    "Created",
 			Status:  metav1.ConditionFalse,
 			Reason:  "Error",
 			Message: fmt.Sprintf("Component create failed: %v", createError),
-		})
+		}
 		logutil.LogAPIResourceChangeEvent(log, component.Name, "Component", logutil.ResourceCreate, createError)
 	}
+	meta.SetStatusCondition(&component.Status.Conditions, condition)
 
 	err := r.Client.Status().Update(ctx, component)
 	if err != nil {
-		log.Error(err, "Unable to update Component")
+		// Retry, and if still fails, then return an error
+		curComponent := &appstudiov1alpha1.Component{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "Application",
```

Comment:
```suggestion
			TypeMeta: metav1.TypeMeta{
				APIVersion: "appstudio.redhat.com/v1alpha1",
				Kind:       "Component",
```
should be? Or do we even need to set TypeMeta?

Reply:
Lol oops.

Should be removed. The typemeta doesn't factor into what's retrieved. This was leftover from testing.

Reply:
Fixed